synfig-core/src/modules/mod_noise/valuenode_random.cpp

```diff
@@ -66,10 +66,14 @@ ValueNode_Random::ValueNode_Random(const ValueBase &value):
 	set_children_vocab(ret);
 	random.set_seed(time(NULL));
 
+	ValueNode_Const::Handle val_smooth;
+	val_smooth = ValueNode_Const::Handle::cast_dynamic(ValueNode_Const::create(int(RandomNoise::SMOOTH_CUBIC)));
```

Comment:
No need to (slow) dynamic cast here: We know it IS a `ValueNode_Const`, because it was created by `ValueNode_Const::create()`.

If a type cast is needed, use `static_cast` .
Same for other cases in this PR.

Reply:
Right :+1: